### ISSUE TITLE
Download link for package archive?

### ISSUE THREAD

**Issue by trusktr** [npm/newww#592]
I don't see a link on package pages for download the package (e.g. a zip or tar file). npm install is having an error, so I'd like to simply download the package manually to see what's going on. Is there some way to download a package without npm install?

**Comment by trusktr** [comment]
I can go to `registry.npmjs.org/package-name` and see the download URLs for the `.tgz` files. A simple download link for the latest version on package pages would be nice.

**Comment by zeke** [comment]
Not a bad idea, but in most cases a downloaded file is not really what you want. I thought about turning the little download-ish icon next to the `npm install foo` text into a link that reveals a message when clicked:

> To install and use this package, you need the npm command-line client. To install npm, go [here]. If you really just want to download the package tarball, it's [here].

What do you think?

**Comment by trusktr** [comment]
Yeah, something like that would be nice! It would make it easier to copy the URL and change the version number.

**Comment by codelion** [comment]
:+1:

**Comment by Klortho** [comment]
+1

**Comment by aredridel** [comment]
I'd love some feedback from everyone on this issue -- what are your use cases here? What makes this a useful thing? Context would make deciding whether to do this or not a lot easier!

**Comment by BenjaminVadant** [comment]
Hi, 
maybe it could be cool/better to be able to download the dependencies as well : 
In my work area, Internet access to the Intranet servers is forbidden, the only way I have to get package is to download the tarball of each package I use and their dependencies, which is quite long and boring.

On a external (personal) server, I've made a little script that's downloading packages and their dependencies and then zip it. You can test it here : http://www.vadant.fr:4000/

What do you think?

**Comment by aredridel** [comment]
Clever, but covers a narrow range of use cases, @BenjaminVadant!

Really, you may want to consider registry replication -- especially via our [npm on-site product](https://www.npmjs.com/npm/on-site) for 'behind the firewall' applications.

**Comment by BenjaminVadant** [comment]
Interesting but way too overkill for my use!

I was thinking of something useable on free wifi, bad connection ... It's more about getting the package you need to work (or continue working) wherever you are. 

Registry replication would mean (for me) a massive use whereas I'm more interested in punctual use.

**Comment by aredridel** [comment]
Ah, I see!

Part of the problem is that if the registry isn't reliable where you are, a download link is going to suffer very similar problems. A single http stream instead of several, but no retries in browsers (and npm the command line tool does retry)
